mlflow/projects/__init__.py

```diff
@@ -185,48 +115,103 @@ def _get_db_hostname_and_auth():
-        env_vars[tracking._EXPERIMENT_ID_ENV_VAR] = experiment_id
     req_body_json = {
-        'run_name': 'MLflow Job Run for %s' % uri,
+        'run_name': 'MLflow Run for %s' % project_uri,
         'new_cluster': cluster_spec,
         'shell_command_task': {
-            'command': _get_databricks_run_cmd(uri, entry_point, version, parameters),
+            'command': command,
             "env_vars": env_vars
         },
         "libraries": [{"pypi": {"package": "mlflow==%s" % VERSION}}]
     }
-    eprint("=== Running entry point %s of project %s on Databricks. ===" % (entry_point, uri))
     run_submit_res = rest_utils.databricks_api_request(
         hostname=hostname, endpoint="jobs/runs/submit", token=token, auth=auth, method="POST",
         req_body_json=req_body_json)
-    run_id = run_submit_res["run_id"]
+    job_run_id = run_submit_res["run_id"]
     eprint("=== Launched MLflow run as Databricks job run with ID %s. Getting run status "
-           "page URL... ===" % run_id)
+           "page URL... ===" % job_run_id)
     run_info = rest_utils.databricks_api_request(
         hostname=hostname, endpoint="jobs/runs/get", token=token, auth=auth, method="GET",
-        params={"run_id": run_id})
+        params={"run_id": job_run_id})
     jobs_page_url = run_info["run_page_url"]
     eprint("=== Check the run's status at %s ===" % jobs_page_url)
+    return job_run_id
+
+
+def _create_databricks_run(experiment_id, source_name, source_version, entry_point_name):
+    # Figure out tracking URI
+    tracking_uri = tracking.get_tracking_uri()
+    if tracking._is_local_uri(tracking_uri):
+        # TODO: we'll actually use the Databricks deployment's tracking URI here in the future
+        eprint("WARNING: MLflow tracking URI is set to a local URI (%s), so results from Databricks"
+               "will not be logged permanently. Performing Databricks Run "
+               "asynchronously." % tracking_uri)
+        return None
+    else:
+        # Assume non-local tracking URIs are accessible from Databricks (won't work for e.g. localhost)
+        # TODO: need to resolve the version to a git commit here (will do that soon)
+        return tracking._create_run(experiment_id=experiment_id,
+                                    source_name=source_name,
+                                    source_version=source_version,
+                                    entry_point_name=entry_point_name,
+                                    source_type=SourceType.PROJECT)
+
+
+def _wait_databricks(databricks_run_id, sleep_interval=30):
+    """ Temporary API for polling Databricks Job run for termination. """
+    hostname, token, username, password, = _get_databricks_hostname_and_auth()
+    auth = (username, password) if username is not None and password is not None else None
+    result_state = None
+    while result_state is None:
+        res = rest_utils.databricks_api_request(
```

Comment:
Should be a bit careful here. If this status request fails (for instance timeout), it will throw from here. If the intension is to block on databricks, we should do a bit of exception handling for [specific http codes: Timeout, Gateway error, N/w Error, Service not available... etc](https://www.restapitutorial.com/httpstatuscodes.html), do N # of retries before bailing.